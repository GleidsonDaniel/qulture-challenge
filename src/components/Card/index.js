import React from 'react';
import {View, StyleSheet, Platform, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

import {
  widthPercentageToDP,
  heightPercentageToDP,
} from '~/helpers/pixelHandler';

const Card = ({
  children,
  elevation,
  opacity,
  cornerRadius,
  backgroundColor,
  onPress,
}) => {
  const cardStyle = Platform.select({
    ios: () =>
      StyleSheet.create({
        card: {
          shadowRadius: elevation,
          shadowOpacity: opacity,
          shadowOffset: {width: 0, height: elevation},
          borderRadius: cornerRadius,
          backgroundColor,
          width: widthPercentageToDP('46%'),
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
    android: () =>
      StyleSheet.create({
        card: {
          elevation,
          borderRadius: cornerRadius,
          backgroundColor,
          width: widthPercentageToDP('46%'),
          height: heightPercentageToDP('28%'),
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
        },
      }),
  })();

  return (
    <TouchableOpacity style={{alignItems: 'center'}} onPress={onPress}>
      <View style={cardStyle.card}>{children}</View>
    </TouchableOpacity>
  );
};

Card.propTypes = {
  backgroundColor: PropTypes.string,
  elevation: PropTypes.number,
  cornerRadius: PropTypes.number,
  opacity: PropTypes.number,
  onPress: PropTypes.func,
  children: PropTypes.element,
};

Card.defaultProps = {
  backgroundColor: '#ffffff',
  elevation: 3,
  cornerRadius: 5,
  opacity: 0.5,
  onPress: () => {},
  children: <View />,
};

export default Card;
