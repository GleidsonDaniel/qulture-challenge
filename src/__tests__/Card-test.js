import React from 'react';
import {shallow} from 'enzyme';
import Card from '~/components/Card';

describe('Card', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Card />);
      expect(component).toMatchSnapshot();
    });
  });
});
