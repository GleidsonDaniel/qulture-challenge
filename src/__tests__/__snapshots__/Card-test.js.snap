// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
